app/controllers/stories_controller.rb

```diff
@@ -36,13 +36,17 @@ def update
     authorize @story
     @story.acting_user = current_user
     @story.base_uri = project_url(@story.project)
+    result = StoryOperations::Update.new.call(story: @story, data: allowed_params, current_user: current_user)
     respond_to do |format|
-      if StoryOperations::Update.call(@story, allowed_params, current_user)
-        format.html { redirect_to project_url(@project) }
-        format.js   { render json: @story }
-      else
-        format.html { render action: 'edit' }
-        format.js   { render json: @story, status: :unprocessable_entity }
+      Dry::Matcher::ResultMatcher.call(result) do |on|
+        on.success do |story|
+          format.html { redirect_to project_url(@project) }
+          format.js   { render json: story }
+        end
+        on.failure do |story|
+          format.html { render action: 'edit' }
+          format.js   { render json: story, status: :unprocessable_entity }
+        end
       end
```

Comment:
```suggestion
     authorize @story
     
     @story.acting_user = current_user
     @story.base_uri = project_url(@story.project)
     
    result = StoryOperations::Update.new.call(story: @story, data: allowed_params, current_user: current_user)
    
    respond_to do |format|
      Dry::Matcher::ResultMatcher.call(result) do |on|
        on.success do |story|
          format.html { redirect_to project_url(@project) }
          format.js   { render json: story }
        end
        on.failure do |story|
          format.html { render action: 'edit' }
          format.js   { render json: story, status: :unprocessable_entity }
        end
      end
```

small suggestion just to give some parts of the code breathing room. Make sense to you? same thing appies for the next block changes.

Reply:
makes sense